src/pages/Admin/Charity/Permissions/useSubmit.ts

```diff
@@ -0,0 +1,85 @@
+import { EncodeObject } from "@cosmjs/proto-signing";
+import { useFormContext } from "react-hook-form";
+import { ProposalMeta } from "pages/Admin/types";
+import { useAdminResources } from "pages/Admin/Guard";
+import { useErrorContext } from "contexts/ErrorContext";
+import { useGetWallet } from "contexts/WalletContext";
+import CW3 from "contracts/CW3";
+import SettingsController from "contracts/SettingsController";
+import useCosmosTxSender from "hooks/useCosmosTxSender";
+import { isEmpty } from "helpers";
+import { getPayloadDiff, getTagPayloads } from "helpers/admin";
+import createUpdateEndowmentControllerMsg from "./createUpdateEndowmentControllerMsg";
+import { FormValues } from "./schema";
+
+export default function useSubmit() {
+  const { id, cw3, propMeta } = useAdminResources<"charity">();
+  const { handleError } = useErrorContext();
+  const {
+    formState: { isSubmitting, errors },
+    handleSubmit,
+    reset,
+    watch,
+  } = useFormContext<FormValues>();
+  const { wallet } = useGetWallet();
+  const sendTx = useCosmosTxSender();
+
+  async function onSubmit({
+    initialValues,
+    userDelegate,
+    ...newValues
+  }: FormValues) {
+    try {
+      const diff = getPayloadDiff(initialValues, newValues);
+
+      if (isEmpty(Object.entries(diff))) {
+        return handleError("No changes detected");
+      }
+
+      const settingsController = new SettingsController(wallet);
+      const updateMsg = createUpdateEndowmentControllerMsg(id, diff);
```

Comment:
bug: having, set all fields to `delegate` , when undelegating `account fees`, some fields are also undelegated like `beneficiary_allow_list`
<img width="1206" alt="image" src="https://user-images.githubusercontent.com/89639563/225831562-5c0c3ab8-1269-4b4d-b951-f5d94c1391f1.png">

REPRO: 
1. go to `admin/27/permissions`
2. check all fields are delegated to address
3. undelegate `account_fees`
4. verify `msg` in keplr wallet prompt 
5. notice `beneficiary_allowlist` is also undelegated

Reply:
This seems to be caused by a bug in `getPayloadDiff`, created https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1947 to address this

Reply:
Was caused by a combination of `getPayloadDiff` bug and a wrong assignment 🤦‍♂️ (see https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1885/commits/bf9367d7c40d641ea068de391ac065a650029020)